components/button/README.md

```diff
@@ -1,28 +1,50 @@
 # Buttons
 
-## d2l-button
-
-The `d2l-button` element can be used just like the native button element, but also supports the `primary` attribute for denoting the primary button.
-
-![Button](./screenshots/button.png?raw=true)
+A Button is used to communicate and perform an action.
 
 ```html
+<!-- docs: demo -->
 <script type="module">
   import '@brightspace-ui/core/components/button/button.js';
+  import '@brightspace-ui/core/components/button/button-subtle.js';
+  import '@brightspace-ui/core/components/button/button-icon.js';
 </script>
-<d2l-button>My Button</d2l-button>
+<d2l-button>Button</d2l-button>
+<d2l-button primary>Primary Button</d2l-button>
+<d2l-button-subtle text="Subtle Button" icon="tier1:gear"></d2l-button-subtle>
+<d2l-button-icon text="Icon Button" icon="tier1:gear"></d2l-button-icon>
 ```
 
-**Properties:**
+## Best Practices
+<!-- docs: start best practices -->
+<!-- docs: start dos -->
+* Use buttons to cause an action or launch a workflow
+* Keep button text short - see "Writing" guidelines
+<!-- docs: end dos -->
+
+<!-- docs: start donts -->
+* Don't use multiple Primary buttons - the primary action should always be clear and obvious
+* Avoid icon-only buttons with unfamiliar icons, only use these for familiar concepts
+* Don't use buttons for navigation, use a link instead
+* Don't open menus with buttons - use a dropdown instead
+<!-- docs: end donts -->
+<!-- docs: end best practices -->
+
+## Button [d2l-button]
+
+The `d2l-button` element can be used just like the native button element, but also supports the `primary` attribute for denoting the primary button.
 
-| Property | Type | Description |
-|--|--|--|
-| `description` | String | A description to be added to the `button` for accessibility |
-| `disabled` | Boolean | Disables the button |
-| `disabledTooltip` | String | Tooltip text when disabled |
-| `primary` | Boolean | Styles the button as a primary button |
```

Comment:
`primary` is located in ButtonMixin which causes it to appear in attributes for Subtle and Icon buttons even though it does nothing for them. I haven't figured out a way around this. We can add `@ignore` and then also put primary in button but that seems hacky and there is at least one other usage of ButtonMixin that uses primary so that could have downstream affects. Another option is add some sort of property exclusion functionality to the daylight site code. Thoughts on this?

Reply:
Hmm, seems like it doesn't belong there eh. Could `primary` be removed from the mixin and then just copied to the places using the mixin that also needed it?

Reply:
We could do that. I believe it's just Button and an IPSIS usage so should be straightforward.